python/python/tests/test_async_client.py

```diff
@@ -1190,6 +1190,29 @@ async def test_sunionstore(self, redis_client: TRedisClient):
             "g",
         }
 
+    @pytest.mark.parametrize("cluster_mode", [True, False])
+    @pytest.mark.parametrize("protocol", [ProtocolVersion.RESP2, ProtocolVersion.RESP3])
+    async def test_sinter(self, redis_client: TRedisClient):
+        key1 = f"{{testKey}}:{get_random_string(10)}"
+        key2 = f"{{testKey}}:{get_random_string(10)}"
+        non_existing_key = f"{{testKey}}:non_existing_key"
+        member1_list = ["a", "b", "c"]
+        member2_list = ["c", "d", "e"]
+
+        # positive test case
+        assert await redis_client.sadd(key1, member1_list) == 3
+        assert await redis_client.sadd(key2, member2_list) == 3
+        assert await redis_client.sinter([key1, key2]) == {"c"}
+
+        # non-existing key returns empty set
+        assert await redis_client.sinter([key1, non_existing_key]) == set()
+
```

Comment:
Sho has asked me before to add a test where you pass an empty key list to the command. The test should assert that the command fails since the key list should not be empty. Can you add that test here? You can use this as an example: https://github.com/aws/glide-for-redis/blob/main/python/python/tests/test_async_client.py#L2099